lib/api/src/engine.rs

```diff
@@ -40,12 +52,13 @@ impl AsEngineRef for EngineRef<'_> {
     }
 }
 
-impl<P> AsEngineRef for P
-where
-    P: Deref,
-    P::Target: AsEngineRef,
-{
-    fn as_engine_ref(&self) -> EngineRef<'_> {
-        (**self).as_engine_ref()
-    }
-}
+// impl<P> AsEngineRef for P
+// where
+//     P: Deref,
+//     P::Target: AsEngineRef,
+// {
+//     fn as_engine_ref(&self) -> EngineRef<'_> {
+//         EngineRef { inner: **self }
+//         (**self).as_engine_ref()
+//     }
+// }
```

Comment:
I might need your help on reenabling this @Michael-F-Bryan

Reply:
You're running into a coherence issue because we have a blanket `impl<P: Deref> EngineRef for P` and also explicitly implement `AsEngineRef` for `Engine` which is a type alias for `engine_imp::Engine`, which is a re-export of `wasmer_compiler::Engine`, which belongs to another crate. 

The problem is our `wasmer_compiler` crate could one day decide to implement `Deref` and return something which itself implements `AsEngineRef`, causing a conflict.

The fix is actually pretty simple, and we were probably going to do it anyway... Create a new `Engine` struct in `lib/api/src/engine.rs` which wraps `engine_imp::Engine` and implement `AsEngineRef` for *that* instead of doing `pub type Engine = engine_imp::Engine`.  That way `Engine` becomes part of the `wasmer` crate again and there are no coherence issues.

Reply:
This is re-enabled now. Closing